### PR TITLE
Fix the "a tag or digest is required" check

### PR DESCRIPTION
**Pull request by 0sewa0**
## Description

Context:
If you use `name.ParseReference(imageUri)` with where only the image is set, no tag or digest, then the default tag will be `latest`.
`latest` is not allowed/used everywhere so we want to disallow providing just the image, a tag or a digest must be provided.

The Bug:
`if taggedRef.TagStr() == name.DefaultTag` would disallow setting `latest` as a tag, which is not the intended behaviour

## How can this be tested?
Run the unit tests, provide in the dynakube an custom image that has no tag or digest set

## Checklist

- ~[] Unit tests have been updated/added~
- [x] PR is labeled accordingly with a single label
- [x] I have read and understood the [contribution guidelines](https://github.com/Dynatrace/dynatrace-operator/blob/main/CONTRIBUTING.md)

### CODE REVIEW

**Review by StefanHauth — DISMISSED** [review]
LATM

File: pkg/controllers/dynakube/version/updater.go [hunk 1]
```diff
@@ -100,7 +100,7 @@ func setImageIDWithDigest(
 	registryClient registry.ImageGetter,
 	imageUri string,
 ) error {
-	ref, err := name.ParseReference(imageUri)
+	ref, err := name.ParseReference(imageUri, name.WithDefaultTag(""))
```
[the remark this answers is not in the document]
Comment (0sewa0) [reply pkg/controllers/dynakube/version/updater.go:103]: gimme sec 😉

[the remark this answers is not in the document]
Comment (StefanHauth) [reply pkg/controllers/dynakube/version/updater.go:103]: AFAIK think this is already covered in a test.

[the remark this answers is not in the document]
Comment (andriisoldatenko) [reply pkg/controllers/dynakube/version/updater.go:103]: If it's covered why do we have a bug?

[the remark this answers is not in the document]
Comment (0sewa0) [reply pkg/controllers/dynakube/version/updater.go:103]: wait a sec 😂 
you will see

[the remark this answers is not in the document]
Comment (0sewa0) [reply pkg/controllers/dynakube/version/updater.go:103]: look at it now

**Review by andriisoldatenko — APPROVED** [review]

**Pull request merged**